tests/lib/tools/tests.pkgs

```diff
@@ -0,0 +1,117 @@
+#!/bin/sh -e
+
+show_help() {
+    echo "usage: tests.pkgs {install,remove} [PACKAGE...]"
+    echo
+    echo "Package names are standardized based on Debian package names"
+    echo "internally, package names are re-mapped to fit the convention"
+    echo "of the used system."
+}
+
+unsupported() {
+    echo "tests.pkgs: cannot manage packages on this system" >&2
+    exit 1
+}
+
+cmd_install() {
+    # This is re-defined by the backend file.
+    unsupported
+}
+
+cmd_remove() {
+    # This is re-defined by the backend file.
+    unsupported
+}
+
+remap_one() {
+    # This may be re-defined by the backend file.
+    echo "$1"
+}
+
+remap_many() {
+    for pkg in "$@"; do
+        remap_one "$pkg"
+    done
+}
+
+import_backend() {
+    case "$1" in
+        ubuntu-core-*)
+            ;;
+        ubuntu-*|debian-*)
+            #shellcheck source=tests/lib/tools/tests.pkgs.apt.sh
+            . "$TESTSLIB/tools/tests.pkgs.apt.sh"
+            ;;
+        fedora-*|centos-*|redhat-*|amazon-*)
+            #shellcheck source=tests/lib/tools/tests.pkgs.dnf-yum.sh
+            . "$TESTSLIB/tools/tests.pkgs.dnf-yum.sh"
+            ;;
+        opensuse-*)
+            #shellcheck source=tests/lib/tools/tests.pkgs.zypper.sh
+            . "$TESTSLIB/tools/tests.pkgs.zypper.sh"
+            ;;
+        arch-*)
+            #shellcheck source=tests/lib/tools/tests.pkgs.pacman.sh
+            . "$TESTSLIB/tools/tests.pkgs.pacman.sh"
+            ;;
+        *)
+            echo "tests.pkgs: cannot import packaging backend $1" >&2
+            exit 1
+            ;;
+    esac
+}
+
+main() {
+    if [ $# -eq 0 ]; then
+        show_help
+        exit 1
+    fi
+
+    action=
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            -h|--help)
+                show_help
+                exit 0
+                ;;
+            --)
+                shift
+                break
+                ;;
+            install|remove)
+                action="$1"
+                shift
+                break  # consume remaining arguments
+                ;;
+            -*)
+                echo "tests.pkgs: unknown option $1" >&2
+                exit 1
+                ;;
+            *)
+                echo "tests.pkgs: unknown command $1" >&2
+                exit 1
+                ;;
+        esac
+    done
+
+    if [ -z "${SPREAD_SYSTEM:-}" ]; then
```

Comment:
Nitpick, I think this could be checked first, before all the args.

Reply:
I kind of wanted the script to respond to --help even outside of our special environment. Not a strong opinion, can adjust.

Reply:
Ah, you're right, I missed --help. I'd leave it as is then.